frontend/dockerui/build.go

```diff
@@ -112,3 +113,26 @@ func (rb *ResultBuilder) EachPlatform(ctx context.Context, fn func(ctx context.C
 	}
 	return eg.Wait()
 }
+
+func (r *mutableResult) addMeta(k string, v []byte) {
+	r.mu.Lock()
+	r.res.AddMeta(k, v)
+	r.mu.Unlock()
+}
+
+func (r *mutableResult) addRef(k string, ref client.Reference) {
+	r.mu.Lock()
+	r.res.AddRef(k, ref)
+	r.mu.Unlock()
+}
+
+func (r *mutableResult) setRef(ref client.Reference) {
+	r.mu.Lock()
+	r.res.SetRef(ref)
+	r.mu.Unlock()
+}
+
+type mutableResult struct {
```

Comment:
Hm, this is a good point.

I'd rather we rework the `Result` type though (making all of the public types into private ones), instead of having wrapper types around it.